File: dotNet/delegation/test/CsvReaderTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using delegation;
using System.IO;

namespace test
{
    [TestClass]
    public class CsvReaderTest
    {
        public static readonly string FILE_NAME = "resources\\GeoIPCountryWhois-cut.csv";

        private StringWriter outContent, errContent;
        private TextWriter originalOut, originalErr;

        [TestInitialize]
        public void Setup()
        {
            outContent = new StringWriter();
            errContent = new StringWriter();
            originalOut = Console.Out;
            Console.SetOut(outContent);
            originalErr = Console.Error;
            Console.SetError(errContent);
        }

        [TestCleanup]
        public void TearDown()
        {
            Console.SetOut(originalOut);
            Console.SetError(originalErr);
        }

        [TestMethod]
        public void testCsvReaderWorksAsExpected()
        {
            var csvReader = new CsvReader();
            csvReader.parse(FILE_NAME);
            Assert.AreEqual(40448, outContent.ToString().Length);
            Assert.AreEqual(0, errContent.ToString().Length);
        }

        [ExpectedException(typeof(FileNotFoundException))]
        [TestMethod]
        public void testCsvReaderThrowsException()
        {
            CsvReader csvReader = new CsvReader();
            csvReader.parse("foo");
        }
    }
}

File: java/src/test/java/no/bouvet/workshop/solid/CsvReaderTest.java
package no.bouvet.workshop.solid;

import org.hamcrest.CoreMatchers;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.io.ByteArrayOutputStream;
import java.io.File;
import java.io.IOException;
import java.io.PrintStream;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class CsvReaderTest {

    public static final String FILE_NAME = "GeoIPCountryWhois-cut.csv";

    private ByteArrayOutputStream outContent, errContent;

    private PrintStream originalOut, originalErr;

    @Before
    public void setUp() {
        outContent = new ByteArrayOutputStream();
        errContent = new ByteArrayOutputStream();
        originalOut = System.out;
        System.setOut(new PrintStream(outContent));
        originalErr = System.err;
        System.setErr(new PrintStream(errContent));
    }

    @After
    public void tearDown() {
        System.setOut(originalOut);
        System.setErr(originalErr);
    }

    @Test
    public void testCsvReaderWorksAsExpected() throws Exception {
        CsvReader csvReader = new CsvReader();
        File file = new File(getClass().getResource("/" + FILE_NAME).getFile());
        csvReader.parse(file);
        assertThat(outContent.toByteArray().length, is(40604));
        assertThat(errContent.toByteArray().length, is(0));
    }

    @Test(expected = IOException.class)
    public void testCsvReaderThrowsException() throws Exception {
        CsvReader csvReader = new CsvReader();
        csvReader.parse(new File("foo"));
    }
}